images/daemon/pkg/daemon/kata_openshift.go

```diff
@@ -0,0 +1,459 @@
+				return isKataUnInstalled, isCrioDropInUnInstalled, err
+			}
+
+			// Storing it locally so that we can avoid one more call to API server further down
+			if kataConfig.Spec.KataConfigPoolSelector != nil {
+				k.KataConfigPoolLabels = kataConfig.Spec.KataConfigPoolSelector.MatchLabels
+			}
+
+			nodeName, err := getNodeName()
+			if err != nil {
+				return isKataUnInstalled, isCrioDropInUnInstalled, err
+			}
+
+			for _, n := range kataConfig.Status.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList {
+				if n == nodeName {
+					isKataUnInstalled = true
+					break
+				}
+			}
+
+			for _, n := range kataConfig.Status.UnInstallationStatus.Completed.CompletedNodesList {
+				if n == nodeName {
+					isCrioDropInUnInstalled = true
+					break
+				}
+			}
+
+			return isKataUnInstalled, isCrioDropInUnInstalled, err
+		}
+	}
+
+	isKataUnInstalled, isCrioDropInUnInstalled, err := k.KataUninstallChecker()
+	if err != nil {
+		return err
+	}
+
+	if isCrioDropInUnInstalled {
+		return nil
+	}
+
+	nodeName, err := getNodeName()
+	if err != nil {
+		return err
+	}
+
+	if !isKataUnInstalled {
+		// Kata binaries need to be uninstalled
+		err = updateKataConfigStatus(k.KataClient, kataConfigResourceName, func(ks *kataTypes.KataConfigStatus) {
+			ks.UnInstallationStatus.InProgress.InProgressNodesCount++
+		})
+
+		if err != nil {
+			return fmt.Errorf("kata is not installed on the node, error updating kataconfig status %+v", err)
+		}
+
+		if k.KataBinaryUnInstaller == nil {
+			k.KataBinaryUnInstaller = uninstallRPMs
+		}
+
+		err = k.KataBinaryUnInstaller(k)
+
+		if err != nil {
+			// kata uninstallation failed. report it.
+			err = updateKataConfigStatus(k.KataClient, kataConfigResourceName, func(ks *kataTypes.KataConfigStatus) {
+				ks.UnInstallationStatus.InProgress.InProgressNodesCount--
+
+				fn, err := getFailedNode(err)
+				if err != nil {
+					return
+				}
+
+				ks.UnInstallationStatus.Failed.FailedNodesList = append(ks.UnInstallationStatus.Failed.FailedNodesList, fn)
+				ks.UnInstallationStatus.Failed.FailedNodesCount = len(ks.UnInstallationStatus.Failed.FailedNodesList)
+			})
+
+			if err != nil {
+				return fmt.Errorf("kata installation failed, error updating kataconfig status %+v", err)
+			}
+
+		}
+		// mark binaries uninstalled
+		err = updateKataConfigStatus(k.KataClient, kataConfigResourceName, func(ks *kataTypes.KataConfigStatus) {
+			ks.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList = append(ks.UnInstallationStatus.InProgress.BinariesUnInstalledNodesList, nodeName)
+		})
+
+		if err != nil {
+			return fmt.Errorf("kata uninstallation succeeded, but error updating kataconfig status %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func doCmd(cmd *exec.Cmd) error {
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	fmt.Println(cmd.String())
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	return nil
+}
+
+func rpmostreeOverrideReplace(rpms string) error {
```

Comment:
Is this used anywhere?

Reply:
No I think we can remove this as we removed all the callers as well. I'll fix this in this PR.